backend/curriculum_tracking/models.py

```diff
@@ -1477,21 +1477,23 @@ class BurndownSnapshot(models.Model):
     def create_snapshot(Cls, user):
         match = Cls.objects.filter(
             user=user,
-            timestamp__gte=timezone.now()
+            timestamp__gte=timezone.now()  
             - timedelta(hours=Cls.MIN_HOURS_BETWEEN_SNAPSHOTS),
         ).first()
 
         if match:
             # there was a snapshot taken recently, no need to store another one
-            return match
-
+            update_create = Cls.objects.update
+        else:
+            update_create = Cls.objects.create
+    
         cards = AgileCard.objects.filter(assignees=user)
         project_cards = cards.filter(content_item__content_type=ContentItem.PROJECT)
 
         complete_cards = cards.filter(status=AgileCard.COMPLETE)
         complete_project_cards = project_cards.filter(status=AgileCard.COMPLETE)
 
-        return Cls.objects.create(
+        return update_create(
```

Comment:
Sometimes this will return an instance and sometimes it will return None.

Can we rather have consistent behaviour? Always returning the instance seems like the best bet.

Reply:
@sheenarbw I am not sure how this will return a none - looking at lines 1484-1488, one of two will be used either `Cls.objects.update` or `Cls.objects.create` - is there something I'm missing?